src/main/java/htsjdk/samtools/util/SnappyLoader.java

```diff
@@ -57,71 +49,47 @@ public SnappyLoader() {
-        Constructor<OutputStream> outputStreamCtor = null;
-        Class<Error> snappyErrorClass = null;
+        this(verbose, verbose);
+    }
 
+    /**
+     * Constructs a new SnappyLoader which will check to see if snappy is available in the JVM/library path.
+     * @param logFailures  if true output a message to stderr if snappy loading fails
+     * @param logSuccesses if true output a message to stderr if snappy loading succeeds
+     */
+    public SnappyLoader(final boolean logFailures, final boolean logSuccesses) {
         if (java.lang.Boolean.valueOf(System.getProperty("snappy.disable", "false"))) {
             System.err.println("Snappy is disabled via system property.");
+            SnappyAvailable = false;
         }
         else {
+            boolean tmpSnappyAvailable = false;
             try {
-                final Class<InputStream> snappyInputStreamClass = (Class<InputStream>)Class.forName("org.xerial.snappy.SnappyInputStream");
-                final Class<OutputStream> snappyOutputStreamClass = (Class<OutputStream>)Class.forName("org.xerial.snappy.SnappyOutputStream");
-                snappyErrorClass = (Class<Error>)Class.forName("org.xerial.snappy.SnappyError");
-                inputStreamCtor = snappyInputStreamClass.getConstructor(InputStream.class);
-                outputStreamCtor = snappyOutputStreamClass.getConstructor(OutputStream.class, Integer.TYPE);
+                final OutputStream test = new SnappyOutputStream(new ByteArrayOutputStream(1000));
+                test.write("Hello World!".getBytes());
+                test.close();
+                tmpSnappyAvailable = true;
+                if (logSuccesses) System.err.println("Snappy successfully loaded.");
             }
-            catch (NoSuchMethodException e) { /* Do nothing. */ }
-            catch (ClassNotFoundException e) { /* Do nothing. */ }
-        }
-
-        this.SnappyInputStreamCtor = inputStreamCtor;
-        this.SnappyOutputStreamCtor = outputStreamCtor;
-
-        if (this.SnappyInputStreamCtor != null && this.SnappyOutputStreamCtor != null) {
-            // Don't try to call any Snappy code until classes have been found via reflection above.
-            boolean tmpSnappyAvailable;
-            try {
-                if (!LoadSnappy.load()) {
-                    if (verbose) System.err.println("Snappy dll failed to load.");
-                    tmpSnappyAvailable = false;
-                }
-                else {
-                    if (verbose) System.err.println("Snappy stream classes loaded.");
-                    tmpSnappyAvailable = true;
-                }
-            } catch (Error e) {
-                if (e.getClass().equals(snappyErrorClass)) {
-                    if (verbose) System.err.println("Snappy dll failed to load: " + e.getMessage());
-                    tmpSnappyAvailable = false;
-                } else {
-                    throw e;
-                }
+            catch (Throwable e) {
```

Comment:
I'm always worried about catching `Throwables` and eating them.  If we could enumerate the expected `Errors` that could be thrown by failure to load native code and then catch those we'd avoid potentially catching unrelated fatal errors.  `UnsatisfiedLinkError` and `SnappyError` comes to mind as the most obvious but I'm not sure if there are others.